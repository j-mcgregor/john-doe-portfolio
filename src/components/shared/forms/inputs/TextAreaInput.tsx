import React from 'react'
import { TextAreaInputProps } from '../../../../types/interfaces/shared'

const TextAreaInput: React.FC<TextAreaInputProps> = ({
    placeholder,
    label,
    name,
    required,
    value,
    onChange,
}) => (
    <div className="form-group">
        {label && <label htmlFor={name}>{label}</label>}
        <textarea
            placeholder={placeholder}
            className="form-control"
            name={name}
            required={required}
            value={value}
            onChange={onChange}
            data-testid={name}
            aria-label={name}
        ></textarea>
    </div>
)

export default TextAreaInput
