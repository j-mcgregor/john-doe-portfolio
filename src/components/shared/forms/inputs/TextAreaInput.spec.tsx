import React from 'react'
import { render, fireEvent } from '@testing-library/react'
import TextAreaInput from './TextAreaInput'

describe('TextAreaInput', () => {
    it('should render a textarea', () => {
        const mockFn = jest.fn()
        const utils = render(
            <TextAreaInput
                label="Test Textarea"
                placeholder="Test"
                name="test"
                value=""
                onChange={mockFn}
                required
            />
        )

        expect(utils.baseElement).toMatchSnapshot()
    })

    it('should trigger the onChange by changing the input', () => {
        const mockFn = jest.fn()
        const utils = render(
            <TextAreaInput
                label="Test Textarea"
                placeholder="Test"
                name="test"
                value=""
                onChange={mockFn}
                required
            />
        )

        const input = utils.getByLabelText('test')
        fireEvent.change(input, { target: { value: '123' } })
        expect(mockFn).toHaveBeenCalledTimes(1)
    })
})
