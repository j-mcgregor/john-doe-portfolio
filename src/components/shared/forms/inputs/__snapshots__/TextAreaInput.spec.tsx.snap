// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextAreaInput should render a textarea 1`] = `
<body>
  <div>
    <div
      class="form-group"
    >
      <label
        for="test"
      >
        Test Textarea
      </label>
      <textarea
        aria-label="test"
        class="form-control"
        data-testid="test"
        name="test"
        placeholder="Test"
        required=""
      />
    </div>
  </div>
</body>
`;
